File: 프로그래머스/lv0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.py
def solution(my_string):
    answer, tmp = [], ""
    for n in my_string:
        try:
            tmp += str(int(n))
        except:
            try:
                answer.append(int(tmp))
                tmp = ""
            except:
                continue
    try:
        return sum(answer) + int(tmp)
    except:
        return sum(answer)

File: 프로그래머스/lv0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.java
class Solution {
    public int solution(String my_string) {
        int answer = 0;
        int num = 0;
        int tmp = 0;
        String[] stringArray = my_string.split("");
        for (String a : stringArray){
            try{
                tmp = Integer.parseInt(a);
                num *= 10;
                num += tmp;
            }
            catch(NumberFormatException e){
                answer += num;
                num = 0;
            }
        }
        return answer += num;
    }
}